### PR TITLE
add presubmit and preiodic jobs for cri proxy

### PR DESCRIPTION
**Pull request by zhifei92**
for https://github.com/kubernetes/kubernetes/pull/127495

### CODE REVIEW

**Comment by zhifei92** [comment]
cc @SergeyKanzhelev

**Comment by zhifei92** [comment]
I'm not sure if it will work well, because I encountered some errors following the ["Running a ProwJob Locally"](https://docs.prow.k8s.io/docs/build-test-update/#running-a-prowjob-locally) tutorial and couldn't find a solution.

error logs:
```
#  CONFIG_PATH=$(realpath test-infra/config/prow/config.yaml) JOB_CONFIG_PATH=$(realpath test-infra/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml) bash pj.sh pull-kubernetes-node-e2e-cri-proxy
job=pull-kubernetes-node-e2e-cri-proxy
CONFIG_PATH=/home/ubuntu3/go/src/k8s.io/test-infra/config/prow/config.yaml
JOB_CONFIG_PATH=/home/ubuntu3/go/src/k8s.io/test-infra/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml
OUT_DIR=/mnt/disks/prowjob-out   (Only used when creating a new kind cluster.)
KIND_CONFIG=   (Only used when creating a new kind cluster.)
NODE_DIR=/mnt/disks/kind-node   (Only used when creating a new kind cluster.)
latest: Pulling from k8s-prow/mkpj
Digest: sha256:6e71c2d9f675bce23136443cd8093fad59ed3344325d305963e89eadb7d3b6d5
Status: Image is up to date for gcr.io/k8s-prow/mkpj:latest
gcr.io/k8s-prow/mkpj:latest
time="2024-09-29T12:31:15Z" level=warning msg="empty -github-token-path, will use anonymous github client"
time="2024-09-29T12:31:15Z" level=info msg="Throttle(0, 0, [])" client=github
PR Number: 127495
time="2024-09-29T12:31:34Z" level=info msg="GetPullRequest(kubernetes-sigs, provider-aws-test-infra, 127495)" client=github
time="2024-09-29T12:31:34Z" level=debug msg="Using GitHub REST API Version: 2022-11-28" client=github
time="2024-09-29T12:31:34Z" level=debug msg="Used token for request" cache-mode= client=github method=GET path=/repos/kubernetes-sigs/provider-aws-test-infra/pulls/127495 throttled=true
time="2024-09-29T12:31:34Z" level=debug msg="Retrying 404" backoff=2s client=github
time="2024-09-29T12:31:36Z" level=debug msg="Using GitHub REST API Version: 2022-11-28" client=github
time="2024-09-29T12:31:37Z" level=debug msg="Used token for request" cache-mode= client=github method=GET path=/repos/kubernetes-sigs/provider-aws-test-infra/pulls/127495 throttled=true
time="2024-09-29T12:31:37Z" level=debug msg="Retrying 404" backoff=4s client=github
time="2024-09-29T12:31:41Z" level=debug msg="Using GitHub REST API Version: 2022-11-28" client=github
time="2024-09-29T12:31:41Z" level=debug msg="Used token for request" cache-mode= client=github method=GET path=/repos/kubernetes-sigs/provider-aws-test-infra/pulls/127495 throttled=true
time="2024-09-29T12:31:41Z" level=debug msg="GetPullRequest(kubernetes-sigs, provider-aws-test-infra, 127495) finished" client=github duration=7.281360014s
time="2024-09-29T12:31:41Z" level=fatal msg="Failed to default PR" error="failed to fetch PullRequest from GitHub: status code 404 not one of [200], body: {\"message\":\"Not Found\",\"documentation_url\":\"https://docs.github.com/rest/pulls/pulls#get-a-pull-request\",\"status\":\"404\"}"
```

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml [hunk 1]
```diff
@@ -874,6 +874,55 @@ periodics:
             cpu: 4
             memory: 6Gi
 
+- name: ci-kubernetes-node-e2e-cri-proxy
+  interval: 24h
+  cluster: k8s-infra-prow-build
+  annotations:
+    testgrid-dashboards: sig-node-release-blocking
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/node-kubelet.yaml:881]: please do not include into release blocking from the beginning

Comment (SergeyKanzhelev) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml:881]: can put into [kubelet](https://testgrid.k8s.io/sig-node-kubelet)

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml [hunk 2]
```diff
@@ -874,6 +874,55 @@ periodics:
             cpu: 4
             memory: 6Gi
 
+- name: ci-kubernetes-node-e2e-cri-proxy
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/node-kubelet.yaml:877]: name it serial

```suggestion
- name: ci-kubernetes-node-e2e-cri-proxy-serial
```

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml [hunk 3]
```diff
@@ -874,6 +874,55 @@ periodics:
...
+          - runner.sh
+          - /workspace/scenarios/kubernetes_e2e.py
+        args:
+          - --node-args=--image-config-file=/home/prow/go/src/k8s.io/test-infra/jobs/e2e_node/containerd/containerd-main/systemd/image-config.yaml
+          - --deployment=node
+          - --gcp-zone=us-west1-b
+          - '--node-test-args=--cri-proxy-enabled=true  --container-runtime-endpoint=unix:///run/containerd/containerd.sock --container-runtime-pid-file= --kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/containerd.service"'
+          - --node-tests=true
+          - --provider=gce
+          - --test_args=--nodes=1 --focus="\[Feature:CriProxy\]" --skip="\[Flaky\]"
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/node-kubelet.yaml:913]: add a comment that those tests are all serial

```suggestion
         # all criproxy tests are serial, but no need to specify it
          - --test_args=--nodes=1 --focus="\[Feature:CriProxy\]" --skip="\[Flaky\]"
```

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 4]
```diff
@@ -4058,3 +4058,54 @@ presubmits:
               requests:
                 cpu: 8
                 memory: 10Gi
+
+    - name: pull-kubernetes-node-e2e-cri-proxy
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:4062]:
```suggestion
    - name: pull-kubernetes-node-e2e-cri-proxy-serial
```

**Comment by zhifei92** [comment]
@SergeyKanzhelev Hello, please help to review this again.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

this looks correct

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 5]
```diff
@@ -4058,3 +4058,55 @@ presubmits:
               requests:
                 cpu: 8
                 memory: 10Gi
+
+    - name: pull-kubernetes-node-e2e-cri-proxy-serial
+      cluster: k8s-infra-prow-build
+      branches:
+        # TODO(releng): Remove once repo default branch has been renamed
+        - master
+        - main
```
Comment (pacoxu) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:4067]: Is this needed for presubmit CI?

**Comment by pacoxu** [comment]
https://kubernetes.slack.com/archives/C1TU9EB9S/p1728358394071729 some change broke the ci-test-infra-label-sync periodic job on Sept 14, 2024.

**Pull request merged**